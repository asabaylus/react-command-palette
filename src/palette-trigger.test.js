/*  eslint
  no-unused-vars: ["error", { "varsIgnorePattern": "^renderer$" }],
  "function-paren-newline":0  */
import { mount } from "enzyme";
import PaletteTrigger from "./palette-trigger";

describe("PaletteTrigger", () => {
  describe("props.trigger", () => {
    it("should accept a custom child component", () => {
      const MyButton = () => <b>foo</b>;
      const paletteTrigger = mount(<PaletteTrigger trigger={<MyButton />} />);
      expect(paletteTrigger.contains(<b>foo</b>)).toBe(true);
      expect(paletteTrigger).toMatchSnapshot();
    });

    it("should accept a custom text string", () => {
      const paletteTrigger = mount(<PaletteTrigger trigger="foo" />);
      expect(paletteTrigger.find("div").text()).toBe("foo");
      expect(paletteTrigger).toMatchSnapshot();
    });

    it("should be accessible", () => {
      const paletteTrigger = mount(<PaletteTrigger trigger="foo" />);
      expect(paletteTrigger.find("div").prop("tabIndex")).toBe(0);
      expect(paletteTrigger.find("div").prop("role")).toBe("button");
    });
  });
});
