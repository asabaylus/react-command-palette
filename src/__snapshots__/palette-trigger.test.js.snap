// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteTrigger props.trigger should accept a custom child component 1`] = `
<PaletteTrigger
  trigger={<MyButton />}
>
  <CustomTriggerComponent
    trigger={<MyButton />}
  >
    <div
      role="button"
      tabIndex={0}
    >
      <MyButton>
        <b>
          foo
        </b>
      </MyButton>
    </div>
  </CustomTriggerComponent>
</PaletteTrigger>
`;

exports[`PaletteTrigger props.trigger should accept a custom text string 1`] = `
<PaletteTrigger
  trigger="foo"
>
  <CustomTriggerComponent
    trigger="foo"
  >
    <div
      role="button"
      tabIndex={0}
    >
      foo
    </div>
  </CustomTriggerComponent>
</PaletteTrigger>
`;
